hack/shellcheck.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+DIR="${1:-.}"
+if [ "$IS_CONTAINER" != "" ]; then
+  if  find "${DIR}" -type f -name '*.sh' -exec shellcheck --format=gcc {} \+
+  then
+    echo "Shell Check Passed"
```

Comment:
I don't think we need to print a success message.  Exiting zero and the lack of warnings should be sufficient feedback.  We also want to exit nonzero of the `find` exits nonzero.  How about:

```sh
find "${DIR}" -type f -name '*.sh' -exec shellcheck --format=gcc {} \+ || exit 1
```

without the `if`?  Even the `|| exit 1` may be optional, because with:

```sh
if [ "$IS_CONTAINER" != "" ]; then
  find "${DIR}" -type f -name '*.sh' -exec shellcheck --format=gcc {} \+
else
  docker run ...
fi
```

the `find` command will be the last command in the script, so the scripts default exit handling will pass it back to the caller.

Reply:
This success message is to get an execution feedback from the pod log in Prow when it passed.